tests/CMakeLists.txt

```diff
@@ -116,6 +119,9 @@ FOREACH(path ${tests})
     ${TEST_${test}_LIBS} ${TEST_${test}_EXT_LIBS}
     dynamic-graph::dynamic-graph)
 
+  TARGET_INCLUDE_DIRECTORIES(${test}
+    SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
```

Comment:
If you add `TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})`, and after that we use `TARGET_LINK_LIBRARIES(${test} ${PROJECT_NAME}), I would think this `TARGET_INCLUDE_DIRECTORIES(${test} SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})` is not required.

Could you test without it ?

Reply:
I did and had to add it for the tests to compile.

Reply:
I agree with @nim65s , it **should** be there. CMake targets are made for that. My guess is that you used *target_link_library* signature without the visibility keyword (*PUBLIC*, *PRIVATE* or *INTERFACE*).

Reply:
Thank you @olivier-stasse